src/core/matter/tables.rs

```diff
@@ -72,10 +66,23 @@ pub(crate) fn hardage(c: char) -> Result<i32> {
         '0' | '4' | '5' | '6' => Ok(2),
         '1' | '2' | '3' | '7' | '8' | '9' => Ok(4),
         '-' => Err(Box::new(Error::UnexpectedCode(
-            "count code start".to_string(),
+            "{{\
+                \"message\":\"count code start\"\
+            }}"
+            .to_string(),
+        ))),
+        '_' => Err(Box::new(Error::UnexpectedCode(
+            "{{\
+                \"message\":\"op code start\"\
+            }}"
+            .to_string(),
         ))),
-        '_' => Err(Box::new(Error::UnexpectedCode("op code start".to_string()))),
-        _ => Err(Box::new(Error::UnknownHardage(c.to_string()))),
+        _ => Err(Box::new(Error::UnknownHardage(format!(
+            "{{\
+                \"message\":\"unknown hardage\",\
+                \"hard\":{c}\
+            }}"
```

Comment:
This is really fragile and the json is unvalidated. I'm going to be pushing a change that uses `serde_json::json!` to make things much easier to maintain

Reply:
Not to mention the gross escaping.